src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs

```diff
@@ -162,6 +169,93 @@ private void ReviewFailedAndThrow(Task task)
                 ExceptionDispatchInfo.Capture(task.Exception.GetBaseException()).Throw();
         }
 
+        /// <summary>
+        /// Recursively prints roots (steps with no dependencies) and their dependents.
+        /// </summary>
+        private string BuildStepDependencyTree(Dictionary<Type, StepWrapper> stepInfoMap)
```

Comment:
Can't we reuse existing code for building dependency trie?

Reply:
Does not look so actually.